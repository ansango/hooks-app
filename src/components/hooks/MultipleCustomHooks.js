import React from "react";
import { useCounter } from "../../hooks/useCounterCopy";
import { useFetch } from "../../hooks/useFetch";
import "./MultipleCustomHooks.css";

export const MultipleCustomHooks = () => {
  const { state: counter, increment } = useCounter(1);
  const url = `https://www.breakingbadapi.com/api/quotes/${counter}`;
  const { data, loading } = useFetch(url);
  const { author, quote } = !!data && data[0];
  return (
    <div>
      <h1>Multiple Custom Hooks</h1>
      <hr />
      {loading ? (
        <div className="alert alert-info text-center">Loading...</div>
      ) : (
        <blockquote className="blockquote text-end">
          <p className="mb-3">{quote}</p>
          <footer className="blockquote-footer">{author}</footer>
        </blockquote>
      )}
      <button className="btn btn-primary" onClick={increment}>
        Next Quote
      </button>
    </div>
  );
};
