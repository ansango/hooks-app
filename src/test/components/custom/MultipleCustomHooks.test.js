import { shallow } from "enzyme";
import { MultipleCustomHooks } from "../../../components/custom/MultipleCustomHooks";

describe("Test in <MultipleCustomHooks/>", () => {
  test("should be rendered", () => {
    const wrapper = shallow(<MultipleCustomHooks />);
    expect(wrapper).toMatchSnapshot();
  });
});
