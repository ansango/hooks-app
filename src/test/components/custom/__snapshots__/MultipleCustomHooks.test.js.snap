// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <MultipleCustomHooks/> should be rendered 1`] = `
<Fragment>
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Next Quote
  </button>
</Fragment>
`;
